src/Database/Transactions/GenericTransactionManager.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Transactions;
+
+use PDO;
+use Tempest\Database\Exceptions\TransactionException;
+use Throwable;
+
+final class GenericTransactionManager implements TransactionManager
+{
+    public function __construct(private PDO $pdo)
+    {
+    }
+
+    public function begin(): void
+    {
+        $transactionBegun = $this->pdo->beginTransaction();
+
+        if (! $transactionBegun) {
```

Comment:
I decided to go with exceptions because the underlying methods already can throw, so it seems more convenient if both kinds of failures would throw

Reply:
Seems reasonable :)